### PR TITLE
[redirect-when-not-logged-in] 

### LINKED ISSUE #70: Redirect When Not Logged in for /articles/:slug
When accessing `/articles/:slug` when not logged in, and the article doesn't exist, the page still displays and there's a bunch of JS errors eg the client isn't handling the errors properly.

This issue is to address this, so the user gets a redirect to login

### PR DESCRIPTION
**Pull request by ebebbington**
Fixed #70 

**Description**

* Changed the order of how checking if user is authed, so we can check properly

### CODE REVIEW

File: src/vue/store/actions.js [hunk 1]
```diff
@@ -16,30 +16,30 @@ const userDefault = {
...
+            .then(async (response) => {
+              console.log("User is authenticated.");
+              await context.dispatch("setUser", response.data.user);
+              resolve();
+            })
+            .catch((error) => {
+              console.log("User has a session, but it's invalid.");
+              console.log(error.response);
+              context.dispatch("unsetUser");
+              reject();
```
Comment (crookse) [inline src/vue/store/actions.js:35]: we should `resolve()` here. the `axios.{method}` call already receives a Promise that `reject`s. that's how the `axios` call ends up in the `catch()` block. If we `reject` again in the `catch()` block, then callers will have to handle `catch()` blocks. essentially what happens is a chain of unnecessary `catch()` blocks will occur. if somewhere in the code we don't handle a `catch()` block, then we'll run into the `Unhandled Promise rejection` error.

File: src/vue/store/actions.js [hunk 2]
```diff
@@ -16,30 +16,30 @@ const userDefault = {
...
+            });
+        } else {
+          console.log("User is not authenticated.");
+          context.dispatch("unsetUser");
+          reject()
```
Comment (crookse) [inline src/vue/store/actions.js:40]: see comment above

File: src/vue/store/actions.js (hunk 1 above)
Comment (crookse) [reply src/vue/store/actions.js:35]: sorry, we should `resolve(undefined)` instead of `reject()`

Comment (crookse) [reply src/vue/store/actions.js:35]: or we can `resolve(false)`

File: src/public/js/_app.js [hunk 3]
```diff
@@ -145,13 +145,17 @@ const router = new VueRouter({
...
 router.beforeEach(async (to, from, next) => {
-  store.dispatch("checkIfUserIsAuthenticated")
-    .then(() => {
-      next();
-    })
-    .catch(() => {
-      next();
-    });
+  if (to.path !== "/login" && to.path !== "/register" && to.path !== "/") {
+    store.dispatch("checkIfUserIsAuthenticated")
```
Comment (crookse) [inline src/public/js/_app.js:149]: once you've addressed the comment below, then this block can be changed to the following:

```javascript
const result = await store.dispatch("checkIfUserIsAuthenticated");
if (!result) {
  router.push("/login");
}
next();
```

File: src/vue/store/actions.js (hunk 1 above)
Comment (ebebbington) [reply src/vue/store/actions.js]: https://github.com/drashland/deno-drash-realworld-example-app/pull/76/commits/d830455eb41feddabd453c65540bcd7cca707ac8

File: src/vue/store/actions.js (hunk 2 above)
Comment (ebebbington) [reply src/vue/store/actions.js]: https://github.com/drashland/deno-drash-realworld-example-app/pull/76/commits/d830455eb41feddabd453c65540bcd7cca707ac8

File: src/public/js/_app.js (hunk 3 above)
Comment (ebebbington) [reply src/public/js/_app.js]: https://github.com/drashland/deno-drash-realworld-example-app/pull/76/commits/d830455eb41feddabd453c65540bcd7cca707ac8

**Pull request merged**